frontend/js/stats.js

```diff
@@ -106,8 +106,22 @@ const fetchAndFillRunData = async (url_params) => {
             }
 
         } else if(item == 'logs' && run_data?.[item] != null) {
-            // textContent does escaping for us
-            document.querySelector("#logs").textContent = run_data[item];
+            const isJsonString = typeof run_data[item] === 'string' &&
+                                 (run_data[item].startsWith('{') || run_data[item].startsWith('['));
+            if (isJsonString) {
+                try {
+                    const logsData = JSON.parse(run_data[item]);
+                    renderLogsInterface(logsData);
+                } catch (e) {
+                    // Fallback to original text if JSON parsing fails
+                    document.querySelector("#logs").textContent = run_data[item];
+                    document.querySelector("#logs").classList.add("plain-text");
```

Comment:
if it is always plaintext, why is it not in the HTML code already?

Reply:
It is not always plaintext. If the new JSON structure is provided, the class will be removed. But I will change it and add the class into the HTML and let JavaScript remove the class if the JSON was inserted successfully.

Reply:
changed the implementation again.  the css class does not exist anymore. in the legacy case the text is just displayed in a `<pre>` html element.